traversal/selector/exploreIndex.go

```diff
@@ -0,0 +1,63 @@
+package selector
+
+import (
+	"fmt"
+
+	ipld "github.com/ipld/go-ipld-prime"
+)
+
+// ExploreIndex traverses a specific index in a list, and applies a next
+// selector to the reached node.
+type ExploreIndex struct {
+	next     Selector      // selector for element we're interested in
+	interest []PathSegment // index of element we're interested in
```

Comment:
This is... a slice that's always of length one?

It's a bit wizard, but we might actually declare this as `interest [1]PathSegment`, and then have the `Interests()` func return `s.interest[:]` -- the latter syntax skooshes it into a slice, while the fixed length declaration means it can get inlined into the struct rather than needing another malloc.

(Might not matter too much.  But I've become a bit of a malloc hawk whenever it's not too much of a stretch to nab the wins.)

Reply:
(... I'm not actually sure if that will survive escape analysis, but I think it's got a good shot.  Even if not, the fixed length is a nice bit of self-documenting, IMO.)

Reply:
I like it. will do.